shared/src/e2e/driver.ts

```
@@ -499,43 +508,44 @@ export class Driver {
             )
         }
 
-        const handlePromise = wait
-            ? this.page
-                  .waitForFunction(findElementMatchingRegexps, typeof wait === 'object' ? wait : {}, tag, regexps)
-                  .catch(err => {
-                      throw notFoundErr(err)
-                  })
-            : this.page.evaluateHandle(findElementMatchingRegexps, tag, regexps)
-
-        const el = (await handlePromise).asElement()
-        if (!el) {
-            throw notFoundErr()
-        }
-        return el
+        return retry(
+            async () => {
+                const handlePromise = wait
+                    ? this.page
+                          .waitForFunction(
+                              findElementMatchingRegexps,
+                              typeof wait === 'object' ? wait : {},
+                              tag,
+                              regexps
+                          )
+                          .catch(err => {
+                              throw notFoundErr(err)
```

Is there a way you could check the error for a name or class to make sure this wouldn't hide other errors, e.g. a lost Puppeteer connection or invalid parameters?